### PR TITLE
#159082379 A User Should Be Able To Create article

### PR DESCRIPTION
**Pull request by abulojoshua1**
### What does this PR do?
- Enables a user to create an article on Authors Haven

### Description of Task to be completed?
- With these new changes, a user is able to create an article on authors haven.

### How should this be manually tested?
- make a post request to the route `/api/articles/`
- The post request made should be in the form ```{
    "article": {
		"title": "THE ARTICLE TITLE",
		"body": "THE ARTICLE BODY",
		"description": "THE ARTICLE DESCRIPTION"
    }
}```

### Any background context you want to provide?
- I created a new django app named `articles`.
### What are the relevant pivotal tracker stories?
- [#159082379](https://www.pivotaltracker.com/story/show/159082379)

### CODE REVIEW

File: authors/apps/articles/renderers.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+        # or something similar), `data` will contain an `errors` key. We want
+        # the default JSONRenderer to handle rendering errors, so we need to
+        # check for this case.
+        errors = data.get('errors', None)
+
+        if errors is not None:
+            pass
+            # As mentioned about, we will let the default JSONRenderer handle
+            # rendering errors.
+            # return super(UserJSONRenderer, self).render(data)
```
Comment (ssewilliam) [inline authors/apps/articles/renderers.py]: @abulojoshua1 
- Is this Commented code
- I think you should return the line 
   `return super(ArticlesJSONRenderer, self).render(data)`
I think what that does is return the actual error if the class's `render()`  itself has an error

File: authors/apps/articles/models.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    published = models.BooleanField(default=False)
+
+    # this takes the time stamp of when the article was made
+    createdAt = models.DateTimeField(auto_now_add=True)
+
+    # This takes the time stamp of when the article was published
+    updatedAt = models.DateTimeField(auto_now=True)
+
+    # This makes identifies the owner of the publication
+    user = models.ForeignKey(User, on_delete=models.CASCADE)
```
Comment (ssewilliam) [inline authors/apps/articles/models.py]: @abulojoshua1 
- You could have also called this `user` as `user_id`

File: authors/apps/articles/tests/tests_views.py [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+class BaseTest(TestCase):
+
+    def setUp(self):
+        """ automatically run the function before running other tests"""
+        self.test_client = Client()
+
+        self.article_to_create = {
+            "article": {
+                "title": "hahahhaha hahaha hahah ahhah",
+                "body": "<p>this is a body that is here hahahhahaha\n <p",
```
Comment (ssewilliam) [inline authors/apps/articles/tests/tests_views.py]: @abulojoshua1 
Can you clarify on why you are passing in raw HTML here

**Review by ssewilliam — COMMENTED** [review]
@abulojoshua1 
- This looks good, it just needs to clean up
- Also you can provide more clarity on what you did

File: authors/apps/articles/tests/tests_views.py (hunk 3 above)
Comment (lytes20) [reply authors/apps/articles/tests/tests_views.py]: i think this is some data to be used in the test and it should be okay to write some raw html @ssewilliam , maybe @abulojoshua1  could also say something

Comment (abulojoshua1) [reply authors/apps/articles/tests/tests_views.py]: I would like to mimic the presence on an article that has HTML attributes like *Bolding*, paragraphs, etc in an article body. But as for now, that for tests.

File: authors/apps/articles/models.py (hunk 2 above)
Comment (abulojoshua1) [reply authors/apps/articles/models.py]: Ok, I will work on that as soon as possible.

File: authors/apps/articles/renderers.py (hunk 1 above)
Comment (abulojoshua1) [reply authors/apps/articles/renderers.py]: Thanks for Identifying that @ssewilliam, Sharp eye there! I will work on it asap.

File: authors/apps/articles/models.py [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+    # this field makes a publication searchable
+    # it is got off a title but should never be the same
+    slug = models.SlugField(db_index=True, max_length=255, unique=True)
+
+    # this makes the publication viewable by all audiences
+    published = models.BooleanField(default=False)
+
+    # this takes the time stamp of when the article was made
+    createdAt = models.DateTimeField(auto_now_add=True)
```
Comment (geofrocker) [inline authors/apps/articles/models.py]: use underscores for variable naming

Comment (abulojoshua1) [reply authors/apps/articles/models.py]: Ok, @geofrocker. I will do that ASAP.

**Pull request merged**